### PR TITLE
Feature/tracefoss 873 assets filtering

### PR DESCRIPTION
**Pull request by ds-ext-sceronik**

### CODE REVIEW

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asbuilt/rest/AssetAsBuiltController.java [hunk 1]
```diff
@@ -166,6 +166,58 @@ public PageResult<AssetAsBuiltResponse> assets(OwnPageable pageable, @QueryParam
...
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ErrorResponse.class))),
+            @ApiResponse(
+                    responseCode = "500",
+                    description = "Internal server error.",
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ErrorResponse.class)))})
+    @GetMapping("/filtered")
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asbuilt/rest/AssetAsBuiltController.java:216]: Do we really need a new endpoint for this? Isn't it enough to just add the filter criteria to the existing ones?

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asplanned/rest/AssetAsPlannedController.java [hunk 2]
```diff
@@ -157,13 +161,65 @@ public PageResult<AssetAsPlannedResponse> assets(OwnPageable pageable, @QueryPar
...
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ErrorResponse.class))),
+            @ApiResponse(
+                    responseCode = "500",
+                    description = "Internal server error.",
+                    content = @Content(
+                            mediaType = "application/json",
+                            schema = @Schema(implementation = ErrorResponse.class)))})
+    @GetMapping("/filtered")
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asplanned/rest/AssetAsPlannedController.java:211]: Same as above

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/request/OwnPageable.java [hunk 3]
```diff
@@ -23,21 +23,27 @@
...
+import static java.util.Objects.isNull;
+
 @Data
 @AllArgsConstructor
 public class OwnPageable {
     private Integer page;
     private Integer size;
     @ArraySchema(arraySchema = @Schema(description = "Content of Assets PageResults", additionalProperties = Schema.AdditionalPropertiesValue.FALSE, example = "manufacturerPartId,desc"), maxItems = Integer.MAX_VALUE)
     private List<String> sort;
+    private List<String> filter;
```
Comment (ds-lcapellino) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/request/OwnPageable.java:46]: I'm not sure if adding a filter to a pageable class is good practice. Maybe create a new class like OwnFilter and add it as query parameter object to the endpoints?

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asbuilt/rest/AssetAsBuiltController.java (hunk 1 above)
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asbuilt/rest/AssetAsBuiltController.java:216]: I was thinking about not breaking existing logic untill FE is adjusted with following sprints. 
owner param is no longer needed in old endpoints and for it to maintain current logic without breaking FE would already need to filter by owner using new SearchCriteria filter

Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/assets/application/asbuilt/rest/AssetAsBuiltController.java:216]: As agreed on chat will change underlying logic accordingly ! :) thanks

**Review by ds-lcapellino — APPROVED** [review]
LGTM

**Review by ds-lcapellino — APPROVED** [review]
LGTM

**Review by ds-lcapellino — APPROVED** [review]
LGTM

**Pull request merged**